Source/HelixToolkit.Wpf/ExtensionMethods/FloatingPointArrayConverters.cs

```diff
@@ -0,0 +1,191 @@
+﻿#if NET6_0_OR_GREATER
+using System.Runtime.Intrinsics.X86;
+using System.Runtime.Intrinsics;
+#endif
+
+namespace HelixToolkit.Wpf;
+
+internal static class FloatingPointArrayConverters
+{
+    public unsafe static void ConvertFloatToDouble<T1, T2>(int arrayCount, T1[] floatArray, T2[] doubleArray)
+    {
+#if NET8_0_OR_GREATER
+        if (Avx512F.IsSupported)
+        {
+            int floatCount = Vector256<float>.Count;
+            int length = arrayCount / floatCount;
+
+#pragma warning disable CS8500
+            fixed (void* floatDataPtr = floatArray)
+            fixed (void* doubleDataPtr = doubleArray)
+#pragma warning restore CS8500
+            {
+                float* floatData = (float*)floatDataPtr;
+                double* doubleData = (double*)doubleDataPtr;
+
+                for (nuint i = 0; i < (uint)length; i++)
+                {
+                    Vector256<float> d = Avx.LoadVector256(floatData);
+                    Vector512<double> r = Avx512F.ConvertToVector512Double(d);
+                    Avx512F.Store(doubleData, r);
+
+                    floatData += floatCount;
+                    doubleData += floatCount;
+                }
+
+                for (int i = length * floatCount; i < arrayCount; i++)
```

Comment:
What does this loop do?

Reply:
Here is an example.
Suppose that `arrayCount` is 27. It is the count of values (float or double) in the arrays.
`int floatCount = Vector256<float>.Count` is 8 (256 / 32). It is the count of values which will be converted with one simd instruction.
`int length = arrayCount / floatCount` is 3. It is the count of simd instructions to execute.
In this example 24 values (3 * 8) will be converted with simd instructions. It remains 3 values (27 - 24) to convert.
The purpose of the loop is converting the remaining values (from `length * floatCount` (24) to `arrayCount` (27) ).

Reply:
Thanks, haven't touch SIMD for a long time, forgot most of this stuff.